app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java

```diff
@@ -0,0 +1,182 @@
+package org.pesmypetcare.mypetcare.utilities;
+
+public class DateTime implements Comparable<DateTime> {
+    public static final int HOUR_START_POS = 11;
+    private static final int DAYS_30 = 30;
+    private static final int DAYS_31 = 31;
+    private static final int FEBRUARY = 2;
+    private static final int JULY = 7;
+    private static final int LEAP_YEAR_FREQ = 4;
+    private static final int TWO_LAST_DIGITS = 100;
+    private static final int DAYS_29 = 29;
+    private static final int DAYS_28 = 28;
+    private static final int DECEMBER = 12;
+    private static final int MAX_HOUR = 24;
+    private static final int MAX_MINUTES_SECONDS = 60;
+    private static final int FIRST_TWO_DIGITS = 10;
+    private static final int MONTH_START_POS = 5;
+    private static final int DAY_START_POS = 8;
+    private static final int HOUR_END_POS = 13;
+    private static final int MIN_START_POS = 14;
+    private static final int MIN_END_POS = 16;
+    private static final int SEC_START_POS = 17;
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int minutes;
+    private int seconds;
+
+    public DateTime(int year, int month, int day, int hour, int minutes, int seconds) throws InvalidFormatException {
+        this.year = year;
+        this.month = month;
+        this.day = day;
+        this.hour = hour;
+        this.minutes = minutes;
+        this.seconds = seconds;
+        if (isNegative(year, month, hour, minutes, seconds) || isOutOfRange(month, hour, minutes, seconds)) {
+            throw new InvalidFormatException();
+        }
+    }
+
+    public DateTime(String dateTime) {
+        this.year = Integer.parseInt(dateTime.substring(0, LEAP_YEAR_FREQ));
+        this.month = Integer.parseInt(dateTime.substring(MONTH_START_POS, JULY));
+        this.day = Integer.parseInt(dateTime.substring(DAY_START_POS, FIRST_TWO_DIGITS));
+        this.hour = Integer.parseInt(dateTime.substring(HOUR_START_POS, HOUR_END_POS));
+        this.minutes = Integer.parseInt(dateTime.substring(MIN_START_POS, MIN_END_POS));
+        this.seconds = Integer.parseInt(dateTime.substring(SEC_START_POS));
+    }
+
+    private boolean isOutOfRange(int month, int hour, int minutes, int seconds) {
+        int nDays = numberOfDays(year, month);
+        boolean dateOutOfRange = month > DECEMBER || day > nDays;
+        boolean hourOutOfRange = hour >= MAX_HOUR || minutes >= MAX_MINUTES_SECONDS || seconds > MAX_MINUTES_SECONDS;
+        return dateOutOfRange || hourOutOfRange;
+    }
+
+    private boolean isNegative(int year, int month, int hour, int minutes, int seconds) {
+        boolean isYearNegative = year < 0 || month < 0 || day < 0;
+        boolean isDateNegative = hour < 0 || minutes < 0 || seconds < 0;
+        return isYearNegative || isDateNegative;
+    }
+
+    private int numberOfDays(int year, int month) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)